dohq_artifactory/admin.py

```diff
@@ -686,34 +690,128 @@ def _read_response(self, response):
         self.includesPattern = response["includesPattern"]
         self.excludesPattern = response["excludesPattern"]
         self._repositories = response.get("repositories", [])
-        if "principals" in response:
-            if "users" in response["principals"]:
-                self._users = response["principals"]["users"]
-            if "groups" in response["principals"]:
-                self._groups = response["principals"]["groups"]
+        self._users = {}
+        self._groups = {}
+        if response.get("principals"):
+            if response["principals"].get("users"):
+                self._users = self._permissions_from_json(
```

Comment:
I agree with adding `_permissions_from_json`, but do not with changing `if something in container` to `if container contains something or give me implicitly None otherwise`. 

The conditional `if 'key' in adict` I'd say more readeable for humans than `if adict.get('key')`

Reply:
`if 'key' in adict` will be false only if key is missing.
`if adict.get('key')` will be false if key is missing OR value by this key is coercible to False (literally False, None or empty container).

Reply:
Fixed